base/compiler/ssair/inlining.jl

```diff
@@ -1483,15 +1487,12 @@ function handle_concrete_result!(cases::Vector{InliningCase}, result::ConcreteRe
     return true
 end
 
+may_inline_concrete_result(result::ConcreteResult) =
+    isdefined(result, :result) && is_inlineable_constant(result.result)
```

Comment:
I am hoping someday we can  get rid of this arbitrary size cutoff entirely, and just do something simple like:
```julia
is_inlineable_constant(x) = (!ismutable(x) && datatype_pointerfree(typeof(x))) || x isa Type || x isa Symbol
```
this avoids embedding pointers to random bits of mutable memory, but otherwise drops the random size cut-off